### ISSUE TITLE
[Bug]: Create Index flow does not lock the Original Table

### ISSUE THREAD

**Issue by arjunsk** [matrixorigin/matrixone#13238]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): main , 1.1-dev, 1.0-dev
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Create Index does not exclusively lock the hidden index table.

### Expected Behavior

During "Dynamic Index Creation", we should have an exclusive lock on the original table to prevent "missing data" in the hidden index table.

**Comment by arjunsk** [comment]
Add this line
https://github.com/matrixorigin/matrixone/blob/35b27e2986f7e5bf039c68301975f6975f7da930/pkg/sql/compile/ddl.go#L129-L131

to 

https://github.com/matrixorigin/matrixone/blob/35b27e2986f7e5bf039c68301975f6975f7da930/pkg/sql/compile/ddl.go#L983

**Comment by arjunsk** [comment]
@qingxinhome I think, this change should be cherry picked to all the 3 branches: main, 1.1-dev and 1.0-dev.

**Comment by arjunsk** [comment]
Waiting for mergify unfreeze

**Comment by arjunsk** [comment]
Merged to main, 1.0 and 1.1
